mlir/include/mlir/IR/OpBase.td

```diff
@@ -2039,6 +2039,8 @@ def ConstantLike : NativeOpTrait<"ConstantLike">;
 def IsolatedFromAbove : NativeOpTrait<"IsIsolatedFromAbove">;
 // Op results are float or vectors/tensors thereof.
 def ResultsAreFloatLike : NativeOpTrait<"ResultsAreFloatLike">;
+// Op results are not the same for each thread.
```

Comment:
If we don't plan to upstream this trait anytime soon, how about defining in polygeist to avoid merge conflict.

Reply:
+1, but IIRC @etiotto plans to attach the trait to ops in upstream dialects, so I fear it makes sense to have it that high in include hierarchy, unless there is a way to dynamically attach traits to existing ops (which I think there isn't).

Reply:
Yes it would, however I wasn't planning to attach the trait to other dialects operations given we do not use those dialects where we need this analysis to run. Noting also that upstreaming the trait without upstreaming the analysis would be problematic (lack of consumers). In order to upstream the analysis we would also need to upstream the reaching def. analysis, the SYCL alias analysis, and likely other changes we have downstream. And that, given current resources and timeline, is not on the critical path.

My pref is to keep the trait as it is in this PR because the footprint is quite small and so the chances of merge conflicts isn't high. However @whitneywhtsang is the one that usually take care of merging MLIR changes so @whitneywhtsang is you really think the merge conflict would make merging harder than it needs to be I will move it to the SYCL dialect

Reply:
I forgot that we may want to use it for upstream dialects, e.g., GPU, so I am ok to keep it here.